src/strategies/DSL/actions/limitSwapAction.ts

```diff
@@ -1,17 +1,44 @@
-import { LimitSwapActionArgs, PrimitiveArgs } from '@brinkninja/types'
+import { toTokenArgs } from '../../../internal/toTokenArgs'
+import { LimitSwapActionArgs, PrimitiveArgs, PrimitiveParamValue } from '@brinkninja/types'
+import { convertToX96HexPrice } from '@brink-sdk/internal/price';
+
+const FLAT_PRICE_CURVE_ADDRESS = '0xC509733B8dDdbab9369A96F6F216d6E59DB3900f';
 
 function limitSwapAction ({
+  id,
   owner,
   tokenIn,
   tokenOut,
   tokenInAmount,
-  tokenOutAmount,
-  price
+  tokenOutAmount
 }: LimitSwapActionArgs): PrimitiveArgs[] {
+  if (tokenOutAmount === undefined) {
+    throw new Error('tokenOutAmount is required')
+  }
+
```

Comment:
not sure what happened to my comment about runtime BigInt checking, but it could be done here to convert tokenOutAmount and tokenInAmount to bigint (and throw if input is not valid for bigint). Then `convertToX96HexPrice` could take bigint params

Reply:
done